ple/games/tetris_engine.py

```diff
@@ -0,0 +1,518 @@
+                     '.O...',
+                     '.....',
+                     '.....']]
+
+I_SHAPE_TEMPLATE = [['..O..',
+                     '..O..',
+                     '..O..',
+                     '..O..',
+                     '.....'],
+                    ['OOOO.',
+                     '.....',
+                     '.....',
+                     '.....',
+                     '.....']]
+
+O_SHAPE_TEMPLATE = [['.OO..',
+                     '.OO..',
+                     '.....',
+                     '.....',
+                     '.....']]
+
+J_SHAPE_TEMPLATE = [['.O...',
+                     '.OOO.',
+                     '.....',
+                     '.....',
+                     '.....'],
+                    ['..OO.',
+                     '..O..',
+                     '..O..',
+                     '.....',
+                     '.....'],
+                    ['.OOO.',
+                     '...O.',
+                     '.....',
+                     '.....',
+                     '.....'],
+                    ['..O..',
+                     '..O..',
+                     '.OO..',
+                     '.....',
+                     '.....']]
+
+L_SHAPE_TEMPLATE = [['...O.',
+                     '.OOO.',
+                     '.....',
+                     '.....',
+                     '.....'],
+                    ['..O..',
+                     '..O..',
+                     '..OO.',
+                     '.....',
+                     '.....'],
+                    ['.OOO.',
+                     '.O...',
+                     '.....',
+                     '.....',
+                     '.....'],
+                    ['.OO..',
+                     '..O..',
+                     '..O..',
+                     '.....',
+                     '.....']]
+
+T_SHAPE_TEMPLATE = [['..O..',
+                     '.OOO.',
+                     '.....',
+                     '.....',
+                     '.....'],
+                    ['..O..',
+                     '..OO.',
+                     '..O..',
+                     '.....',
+                     '.....'],
+                    ['.OOO.',
+                     '..O..',
+                     '.....',
+                     '.....',
+                     '.....'],
+                    ['..O..',
+                     '.OO..',
+                     '..O..',
+                     '.....',
+                     '.....']]
+
+PIECES = {'S': S_SHAPE_TEMPLATE,
+          'Z': Z_SHAPE_TEMPLATE,
+          'J': J_SHAPE_TEMPLATE,
+          'L': L_SHAPE_TEMPLATE,
+          'I': I_SHAPE_TEMPLATE,
+          'O': O_SHAPE_TEMPLATE,
+          'T': T_SHAPE_TEMPLATE}
+
+
+class GameState:
+    def __init__(self):
+        global FPSCLOCK, DISPLAYSURF, BASICFONT, BIGFONT
+        pygame.init()
+        FPSCLOCK = pygame.time.Clock()
+        DISPLAYSURF = pygame.display.set_mode((WINDOWWIDTH, WINDOWHEIGHT))
+        BASICFONT = pygame.font.Font('freesansbold.ttf', 18)
```

Comment:
General comment: I found having text drawn to the screen slows rendering you can choose to omit it if you wish. We still get the reward from the game so I do not believe much is lost.